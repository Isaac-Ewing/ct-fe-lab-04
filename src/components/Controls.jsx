import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ url, rawJson, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit} onChange={onChange} data-testid="controls-form">
      <input type="text" aria-label="urlTextbox" placeholder="URL" value={url} name="url"/>
      <section aria-label="methodSelector">
        <label htmlFor="GET">
          <input type="radio" name="method" value="GET" id="GET"/>
          <div>GET</div>
        </label>
        <label htmlFor="POST">
          <input type="radio" name="method" value="POST" id="POST"/>
          <div>POST</div>
        </label>
        <label htmlFor="PUT">
          <input type="radio" name="method" value="PUT" id="PUT"/>
          <div>PUT</div>
        </label>
        <label htmlFor="PATCH">
          <input type="radio" name="method" value="PATCH" id="PATCH"/>
          <div>PATCH</div>
        </label>
        <label htmlFor="DELETE">
          <input type="radio" name="method" value="DELETE" id="DELETE"/>
          <div>DELETE</div>
        </label>
        <button aria-label="submit-button">Submit</button>
      </section>
      <input type="text" aria-label="jsonText" placeholder="JSON body" value={rawJson} name="rawJson" />
    </form>
  </div>
);

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  rawJson: PropTypes.string.isRequired, 
  onChange: PropTypes.func.isRequired, 
  onSubmit: PropTypes.func.isRequired
};

export default Controls;
