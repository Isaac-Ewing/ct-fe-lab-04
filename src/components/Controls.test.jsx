import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('Controls', () => {
  it('tests for form with two inputs, radio selections, and a button', () => {
    render(
      <Controls 
        valueUrl=""
        valueRawJson=""
        onChange={() => {}}
        onSubmit={() => {}}
      />
    );

    const controlsForm = screen.getByTestId('controls-form');
    expect(controlsForm).toMatchSnapshot();
  });
});
