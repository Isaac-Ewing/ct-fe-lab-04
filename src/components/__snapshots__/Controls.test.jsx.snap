// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls tests for form with two inputs, radio selections, and a button 1`] = `
<form
  data-testid="controls-form"
>
  <input
    aria-label="urlTextbox"
    name="url"
    placeholder="URL"
    type="text"
    value=""
  />
  <section
    aria-label="methodSelector"
    class="radio-route"
  >
    <input
      id="GET"
      name="method"
      type="radio"
      value="GET"
    />
    <label
      for="GET"
    >
      GET
    </label>
    <input
      id="POST"
      name="method"
      type="radio"
      value="POST"
    />
    <label
      for="POST"
    >
      POST
    </label>
    <input
      id="PUT"
      name="method"
      type="radio"
      value="PUT"
    />
    <label
      for="PUT"
    >
      PUT
    </label>
    <input
      id="PATCH"
      name="method"
      type="radio"
      value="PATCH"
    />
    <label
      for="PATCH"
    >
      PATCH
    </label>
    <input
      id="DELETE"
      name="method"
      type="radio"
      value="DELETE"
    />
    <label
      for="DELETE"
    >
      DELETE
    </label>
    <button
      aria-label="submit-button"
    >
      Go!
    </button>
  </section>
  <textarea
    aria-label="jsonText"
    name="rawJson"
    placeholder="JSON body"
    type="text"
  />
</form>
`;
